File: problems/1663_smallest_string_with_a_given_numeric_value/solution.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        result = ['a'] * n
        k -= n
        for i in range(n - 1, -1, -1):
            add = min(25, k)
            result[i] = chr(ord('a') + add)
            k -= add
            if k == 0:
                break
        return ''.join(result)

File: problems/1663_smallest_string_with_a_given_numeric_value/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string getSmallestString(int n, int k) {
        string result(n, 'a');
        k -= n;
        for (int i = n - 1; i >= 0 && k > 0; i--) {
            int add = min(25, k);
            result[i] = 'a' + add;
            k -= add;
        }
        return result;
    }
};
